spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -1557,918 +1624,935 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
-          createBinaryExpr(
-            expr.children(0),
-            expr.children(1),
-            inputs,
-            (builder, binaryExpr) => builder.setArrayContains(binaryExpr))
-        case _ if expr.prettyName == "array_append" =>
-          createBinaryExpr(
-            expr.children(0),
-            expr.children(1),
-            inputs,
-            (builder, binaryExpr) => builder.setArrayAppend(binaryExpr))
-        case _ if expr.prettyName == "array_intersect" =>
-          createBinaryExpr(
-            expr.children(0),
-            expr.children(1),
-            inputs,
-            (builder, binaryExpr) => builder.setArrayIntersect(binaryExpr))
-        case _ =>
-          withInfo(expr, s"${expr.prettyName} is not supported", expr.children: _*)
+              .setListExtract(arrayExtractBuilder)
+              .build())
+        } else {
+          withInfo(expr, "unsupported arguments for ElementAt", child, ordinal)
           None
-      }
-    }
+        }
 
-    /**
-     * Creates a UnaryExpr by calling exprToProtoInternal for the provided child expression and
-     * then invokes the supplied function to wrap this UnaryExpr in a top-level Expr.
-     *
-     * @param child
-     *   Spark expression
-     * @param inputs
-     *   Inputs to the expression
-     * @param f
-     *   Function that accepts an Expr.Builder and a UnaryExpr and builds the specific top-level
-     *   Expr
-     * @return
-     *   Some(Expr) or None if not supported
-     */
-    def createUnaryExpr(
-        child: Expression,
-        inputs: Seq[Attribute],
-        f: (ExprOuterClass.Expr.Builder, ExprOuterClass.UnaryExpr) => ExprOuterClass.Expr.Builder)
-        : Option[ExprOuterClass.Expr] = {
-      val childExpr = exprToProtoInternal(child, inputs)
-      if (childExpr.isDefined) {
-        // create the generic UnaryExpr message
-        val inner = ExprOuterClass.UnaryExpr
-          .newBuilder()
-          .setChild(childExpr.get)
-          .build()
-        // call the user-supplied function to wrap UnaryExpr in a top-level Expr
-        // such as Expr.IsNull or Expr.IsNotNull
-        Some(
-          f(
+      case GetArrayStructFields(child, _, ordinal, _, _) =>
+        val childExpr = exprToProto(child, input, binding)
+
+        if (childExpr.isDefined) {
+          val arrayStructFieldsBuilder = ExprOuterClass.GetArrayStructFields
+            .newBuilder()
+            .setChild(childExpr.get)
+            .setOrdinal(ordinal)
+
+          Some(
             ExprOuterClass.Expr
-              .newBuilder(),
-            inner).build())
-      } else {
-        withInfo(expr, child)
+              .newBuilder()
+              .setGetArrayStructFields(arrayStructFieldsBuilder)
+              .build())
+        } else {
+          withInfo(expr, "unsupported arguments for GetArrayStructFields", child)
+          None
+        }
+      case expr: ArrayRemove => CometArrayRemove.convert(expr, input, binding)
```

Comment:
Although there is a big diff, I suppose they are just code moving and this is the only change?

Reply:
Yes, there are no functional changes. I just moved the inner methods to the top-level and that required adding extra parameters in some bases (such as `input` and `binding`).